src/main/java/core/basesyntax/service/operations/SupplyOperation.java

```diff
@@ -0,0 +1,20 @@
+package core.basesyntax.service.operations;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.model.FruitTransaction;
+import java.util.List;
+
+public class SupplyOperation implements OperationHandler {
+    @Override
+    public void getOperation(List<FruitTransaction> transactions) {
+        List<FruitTransaction> transactionsSelected = transactions.stream()
+                .filter(a -> a.getOperation() == FruitTransaction.Operation.SUPPLY)
+                .toList();
+
+        for (FruitTransaction fruitTransaction : transactionsSelected) {
+            int newValue = Storage.fruits.get(fruitTransaction.getFruit())
+                    + fruitTransaction.getQuantity();
+            Storage.fruits.put(fruitTransaction.getFruit(), newValue);

```

Comment:
Direct access to the `Storage.fruits` map without checking if the key exists can lead to a NullPointerException if the fruit is not already in the map. You should check if the fruit exists in the storage and handle the case where it does not.